packages/react-components/react-spinbutton/src/components/SpinButton/useSpinButton.tsx

```diff
@@ -169,7 +169,7 @@ export const useSpinButton_unstable = (props: SpinButtonProps, ref: React.Ref<HT
 
     let newValue = val + stepSize * dir;
     if (!Number.isNaN(newValue)) {
-      newValue = clampWhenInRange(val, newValue, min, max);
+      newValue = clamp(newValue, min, max);
```

Comment:
> newValue = clamp(newValue, min, max); [](http://example.com/codeflow?start=5&length=38)

The only interesting case that comes to mind that should be considered:

Do you expect a `SpinButton` consumer might want to throw error UI/message/sound to the user when they are at the bounds?

Not sure if that's even supported prior to this change (assuming the value was always within range) since the newValue will not be changing, or if it's a scenario you want the `SpinButton` to support

Reply:
This is a good scenario to consider. I'm going to discuss it more with the team as I think we'd want similar behavior to be applied to other controls should we make this change.

Reply:
I'm going to leave the behavior as-is right now. The [native `input type="number"` behavior](https://codepen.io/seanms/pen/VwQprGX?editors=1011) is consistent with `SpinButton`'s behavior and I think we'd want to take some time to consider similar situations with other controls (e.g., what happens when you get to the end of a Dropdown list?) so we can provide a consistent experience.

It's possible today to use the [slots API](https://master--61b7c406fa7cd4003a8a07ec.chromatic.com/?path=/docs/concepts-developer-customizing-components-with-slots--page) to listen for `onClick` events on SpinButton buttons and `onKeydown` events on Spinbutton input to implement a specific action when SpinButton is "changed" at a bound.

I've opened a [feature request issue](https://github.com/microsoft/fluentui/issues/23043) to consider adding it later.